util/parsec_project.py

```
@@ -187,32 +206,35 @@ def get_functions_in_topological_order(self, reverse_order: bool = False) -> lis
+                result = subprocess.run(cmd, shell=True, capture_output=True, text=True, cwd=path)
         except subprocess.CalledProcessError as e:
-            raise Exception("Error running parsec.") from e
+            raise RuntimeError("Error running parsec") from e
+
+        if not result:
+            msg = f"Failed to run parsec due to an unrecognized run mode = {run_mode}"
+            raise RuntimeError(msg)
 
         path_to_result = Path("analysis.json")
         if not path_to_result.exists():
             raise Exception("parsec failed to produce an analysis")
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)